fmskill/plot.py

```diff
@@ -310,13 +316,24 @@ def scatter(
             markersize=options.plot.scatter.quantiles.markersize,
             **settings.get_option("plot.scatter.quantiles.kwargs"),
         )
-        plt.plot(
-            x_trend,
-            intercept + slope * x_trend,
-            **settings.get_option("plot.scatter.reg_line.kwargs"),
-            label=reglabel,
-            zorder=2,
-        )
+        
+        if fit_to_quantiles: 
+            plt.plot(
```

Comment:
Try to reduce duplication, e.g. by creating new variables that for the x,ys that will be used in the regression. 

Right now, in order to understand the code, you have to compare the two branches of the condition, and try to spot the differences, it is better if it is obvious.

Reply:
Good catch, thanks! I made it more concise